src/lib/Zikula/Bundle/CoreBundle/EventListener/ThemeListener.php

```diff
@@ -161,16 +164,63 @@ public function setUpThemePathOverrides(FilterControllerEvent $event)
         }
     }
 
+    /**
+     * Read the controller annotations and change theme if the annotation indicate that need
+     * @param FilterControllerEvent $event
+     */
+    public function readControllerAnnotations(FilterControllerEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            // prevents calling this for controller usage within a template or elsewhere
+            return;
+        }
+        $controller = $event->getController();
+        list($controller, $method) = $controller;
+        $this->changeThemeByAnnotation($controller, $method);
+    }
+
+    /**
+     * Change a theme based on the annotation
+     * @param $controller
+     * @param $method
+     * @return array|bool|string
+     */
+    public function changeThemeByAnnotation($controller, $method)
```

Comment:
re: moving the Listener class - looking through Symfony, typically the EventListeners are stored in the Bundles (3 of 5 Bundles have a directory). Components typically do *not* have an EventListener directory but an exception is the HttpKernel Component which has many Listeners.

The Zikula/Core isn't organized as **Components** specifically so the comparison isn't 100% true. It is just simply a collection of classes.

The reason I would consider moving the class is to further encapsulate (I'm not sure if that is the proper use of this term here) the Theme classes into one directory.